stl/inc/xstring

```diff
@@ -310,7 +384,7 @@ struct _Narrow_char_traits { // Implement char_traits for narrow character types
 #if _HAS_U8_INTRINSICS

```

Comment:
Ditto "this entire function body could as well be `return __builtin_strlen(_First);`

Reply:
This one appears to be justified. While `memcmp` takes `const void *`, `strlen` takes `const char *`.